pylint/checkers/typecheck.py

```diff
@@ -1901,14 +1902,55 @@ def _detect_unsupported_alternative_union_syntax(self, node: nodes.BinOp) -> Non
             if not allowed_nested_syntax:
                 self._check_unsupported_alternative_union_syntax(node)
 
+    def _includes_version_compatible_overload(self, attrs: list[nodes.NodeNG]):
+        """Check if a set of overloads of an operator includes one that
+        can be relied upon for our configured Python version.
+
+        If we are running under a Python 3.10+ runtime but configured for
+        pre-3.10 compatibility then Astroid will ahve inferred the
+        existence of __or__ / __ror__ on builtins.type, but these aren't
+        available in the configured version of Python.
+        """
+        is_py310_builtin = all(
+            isinstance(attr, (nodes.FunctionDef, astroid.BoundMethod))
+            and attr.parent.qname() == "builtins.type"
+            for attr in attrs
+        )
+        return (not is_py310_builtin) or self._py310_plus
```

Comment:
```suggestion
        return not is_py310_builtin or self._py310_plus
```
The brackets are redundant. `not` binds stronger than `or`.